### ISSUE TITLE
if dismissOnScroll at false, the click on the close button do not close the banner

### ISSUE THREAD

**Issue by F4b1n0u** [buildo/react-cookie-banner#16]
it's the setState() into removeOnScrollListener who trigger the getBanner into the render
witch make disappear it once click on the close button.
but if a props dismissOnScroll at false, listeningScroll is at false as well
and that skips the removeOnScrollListener call
so no render, so the banner do not disappear.

**Comment by gabro** [comment]
I'm not sure I understand.

Can you describe the steps to reproduce this issue and the expected outcome?

**Comment by F4b1n0u** [comment]
if you put the props dismissOnScroll at false, the close button does not work anymore

**Comment by F4b1n0u** [comment]
solved by adding a dummy setState at the end of the onAccept function
but I can't push because of a 403

**Comment by gabro** [comment]
oh I see. Yes, there should be either a dummy `setState` or a `forceUpdate` call.

Good catch!

I'll fix it soon

**Comment by F4b1n0u** [comment]
I'm in favour of the setState :)
forceUpdate sounds like a hack

**Comment by gabro** [comment]
calling an empty `setState` is equally an hack. `forceUpdate` is simply more explicit :)

**Comment by F4b1n0u** [comment]
fair enough :)

**Comment by F4b1n0u** [comment]
and why not use something like:

```javascript
  getInitialState: function(){
    return {
      ...
      displayBanner: this.hasAcceptedCookies()
      ...
    };
  },
  onAccept: function() {
    ...
    this.setState({
      displayBanner: true;
    })
  },
  getBanner: function() {
    ...
    if(  this.hasAcceptedCookies() ) {
      return null;
    }
    ...
  },
  render: function() {
    var displayBanner = this.state.displayBanner;
    ....
    return displayBanner ? this.getBanner();
  }
```

**Comment by gabro** [comment]
@F4b1n0u fixed and released as 0.0.13. Thanks for the report!
